src/util/features_extraction.py

```
@@ -195,75 +199,52 @@ def extract_match(pattern: str, text: str) -> str:
-    
-    # Преобразуем результат в DataFrame
-    features_df = DataFrame(features.tolist())
-    
-    # Объединяем исходный DataFrame с новыми признаками
-    result_df = concat([nomenclatures, features_df], axis=1)
-    
-    return result_df
+    # Применяем регулярные выражения для извлечения характеристик
+    for param_name, pattern in FEATURES_REGEX_PATTERNS.items():
+        nomenclatures[param_name] = nomenclatures['name'].progress_apply(
+            lambda nom_name: extract_match(pattern, nom_name)
```

Ruff (0.8.2)

Function definition does not bind loop variable `pattern`

(B023)